pkg/controller/connectivitycheck/connectivity_check_template.go

```diff
@@ -28,24 +20,47 @@ func NewPodNetworkConnectivityCheckTemplate(address, namespace string, options .
 
 // WithTLSClientCert option specifies the name of the secret in the check namespace that
 // contains a tls client certificate (and key) to use when performing the check.
-func WithTLSClientCert(secretName string) func(*v1alpha1.PodNetworkConnectivityCheck) {
-	return func(check *v1alpha1.PodNetworkConnectivityCheck) {
+func WithTLSClientCert(secretName string) func(*applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration) {
+	return func(check *applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration) {
 		if len(secretName) > 0 {
-			check.Spec.TLSClientCert = v1.SecretNameReference{Name: secretName}
+			check.Spec.TLSClientCert = &v1.SecretNameReference{Name: secretName}
 		}
 	}
 }
 
 // WithSource option replaces the $(SOURCE) token in the name.
-func WithSource(source string) func(*v1alpha1.PodNetworkConnectivityCheck) {
-	return func(check *v1alpha1.PodNetworkConnectivityCheck) {
-		check.Name = strings.Replace(check.Name, "$(SOURCE)", source, -1)
+func WithSource(source string) func(*applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration) {
+	return func(check *applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration) {
+		name := strings.Replace(*check.Name, "$(SOURCE)", source, -1)
+		check.Name = &name
 	}
 }
 
 // WithTarget option replaces the $(TARGET) token in the name.
-func WithTarget(target string) func(*v1alpha1.PodNetworkConnectivityCheck) {
-	return func(check *v1alpha1.PodNetworkConnectivityCheck) {
-		check.Name = strings.Replace(check.Name, "$(TARGET)", target, -1)
+func WithTarget(target string) func(*applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration) {
+	return func(check *applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration) {
+		name := strings.Replace(*check.Name, "$(TARGET)", target, -1)
+		check.Name = &name
+	}
+}
+
+// DeepCopy returns clone for given check object. Now this method considers copying only
+// relevant fields used by check controller.
+func DeepCopy(check *applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration) *applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration {
```

Comment:
This function is not really doing a deep copy, can we rename it to something more fitting?
Also, does it need to be exported?

Reply:
changed the function name and made it private. is this ok now ?